ecommerce/CheckoutController.php

```diff
@@ -105,12 +112,40 @@ public static function create_checkout_page() {
 	}
 
 	/**
-	 * Restrict checkout page
+	 * Pay now ajax handler
 	 *
 	 * @since 3.0.0
 	 *
 	 * @return void
 	 */
+	public function ajax_pay_now() {
+		tutor_utils()->check_nonce();
+
+		$request = Input::sanitize_array( $_POST );
+
+		$course_ids  = $request['course_ids'] ?? '';
+		$coupon_code = $request['coupon_code'] ?? '';
+
+		if ( empty( $course_ids ) ) {
+			$this->json_response(
+				__( 'Invalid cart items' ),
+				'',
+				HttpHelper::STATUS_BAD_REQUEST
+			);
+		}
+
+		$coupon_model  = new CouponModel();
+		$course_ids    = explode( ',', $course_ids );
```

Comment:
Suggested to add another line
$course_ids = array_filter( $course_ids, 'is_numeric' );